CMakeLists.txt

```diff
@@ -275,7 +275,7 @@ configure_file (
   )
 install(SCRIPT tcmu.conf_install.cmake)
 
-install(FILES libtcmu.h libtcmu_common.h tcmu-runner.h
+install(FILES libtcmu.h libtcmu_common.h libtcmu_config.h libtcmu_log.h darray.h
```

Comment:
What was the reason for needing darray.h?

Do we really want to export the config and log stuff to lib users?

If we do then I think libtcmu_config.h could be internal to libtcmu. The tcmu_setup/destory_config calls could go in tcmulib_initialize/close so the lib user does not have to worry about it. For tcmu-runner:main.c, consumer and sync you will need to move some code around.

For libtcmu_log.h just export tcmu_dev_* and tcmu_err/warn/info/dbg/tcmu_dbg_scsi_cmd, and other functions that you think users need now. For example, tcmu_logdir_resetup does not need to be exported if the config code is now internal. tcmu_setup_log/tcmu_logdir_destroy would need to be exported if we cannot figure out a nice way to move that into tcmulib_initialize/tcmulib_close.

Reply:
OK, let's drop them in this version.  Great thanks for reviewing.